File: app/src/main/java/de/mchllngr/androidboilerplatekotlin/module/main/MainPresenter.java
package de.mchllngr.androidboilerplatekotlin.module.main;

import de.mchllngr.androidboilerplatekotlin.base.BasePresenter;

/**
 * {@link com.hannesdorfmann.mosby.mvp.MvpPresenter} for the {@link MainActivity}
 */
@SuppressWarnings("ConstantConditions")
public class MainPresenter extends BasePresenter<MainView> {

//    /**
//     * {@link Example}-class for an example for Dagger2-injection.
//     */
//    @Inject
//    Example exampleDependency;

    @Override
    public void attachView(MainView view) {
        super.attachView(view);
        getApplicationComponent().inject(this);
    }

    @Override
    public void detachView(boolean retainInstance) {
        super.detachView(retainInstance);
    }
}

File: app/src/main/java/de/mchllngr/androidboilerplatekotlin/module/main/MainPresenter.kt
package de.mchllngr.androidboilerplatekotlin.module.main

import de.mchllngr.androidboilerplatekotlin.base.BasePresenter

/**
 * [com.hannesdorfmann.mosby3.mvp.MvpPresenter] for the [MainActivity]
 */
class MainPresenter : BasePresenter<MainView>() {

//    /**
//     * [Example]-class for an example for Dagger2-injection.
//     */
//    @Inject
//    lateinit var example: Example

    override fun attachView(view: MainView) {
        super.attachView(view)
        applicationComponent.inject(this)
    }

    override fun detachView(retainInstance: Boolean) {
        super.detachView(retainInstance)
    }
}
